### PR TITLE
Refactor & replace dashboard Admin API, implement commands for `auth`, `project`, `organization`, and `api-key`

### PR DESCRIPTION
**Pull request by austin-denoble**
## Problem
We need to support admin API functionality in the CLI. Previously, there were commands implemented for working with projects, organizations, and API keys. The original implementation for these features in 2024 used a custom implementation of the Pinecone dashboard APIs which can be found in the `dashboard` package: https://github.com/pinecone-io/cli/tree/main/internal/pkg/dashboard

These commands were removed from the CLI at that point because we couldn't rely on a custom implementation of the dashboard APIs, and we also wanted to limit their usage to the dashboard itself. Ultimately, for maintainability moving forward we would want to rely on specced and versioned APIs, supported through the Go SDK.

## Solution
This PR adds several new commands to the root level of the CLI. The help documentation for each command can be checked by providing the `--help | -h` flags at any level. The commands are focused on authorization, and admin resource management:
- `auth` - groups commands related to logging in and out, setting the target context, and providing service account details.
- `project` - allows working with CRUD operations for projects.
- `api-key` - allows working with CRUD operations for API keys within projects.
- `organization` - allows for describing, updating, listing, and deleting - we don't support organization create via the API yet.

For reviewing the command functionality specifically, look in the folders under `internal/pkg/cli/command/` to see where commands and their sub-commands are grouped. The command names should correspond with the top level folders.

To support adding these new commands leveraging the Go SDK, there were other changes that needed to be made to the CLI:
- Allowing the user to configure `client_id` and `client_secret` for a service account. Since the number of commands related to working with credentials has grown, I grouped things under the `auth` command mentioned above. This included aliasing the existing `login`, `logout`, `target`, and `whoami` commands under the new `auth` command. They are still available at the root. The commands with complex run logic (such as `login`) have had the internal functions extracted into a Run function which can be used in multiple places (see: `internal/pkg/utils/login/login.go`).
- Refactoring things in `internal/pkg/utils/sdk/client.go` to support instantiating a `pinecone.AdminClient` to be used for admin operations. There were also changes needed to `NewPineconeClientForProjectById`, primarily adding a new helper function `getCLIAPIKeyForProject`, which allows the CLI to manage its own API keys for each project it has access to. Since we no longer allow fetching API key secrets after the `create` flow, the CLI needs some way to track it's own keys internally, and use them for non-admin operations. This is similar to what the console does when creating an initial API key for use.
- Made updates to our secrets store (`internal/pkg/utils/configuration/secrets/secrets.go`) to store `ClientId`, `ClientSecret`, and `ProjectAPIKeys`. The `ProjectAPIKeys` store is the one which handles mapping project <-> CLI-specific API key so that the CLI can work with control and data plane operations within that project as needed.

Example help output for root + new commands:

| COMMAND       | `--help` OUTPUT                                                            |
|----------------|----------------------------------------------------|
| `pc --help`          | Screenshot 2025-08-26 at 1 24 37 AM |
| `pc auth --help`  | Screenshot 2025-08-26 at 1 24 58 AM |
| `pc project --help`  | Screenshot 2025-08-26 at 1 25 19 AM |
| `pc organization --help`  | Screenshot 2025-08-26 at 1 25 40 AM |
| `pc api-key --help`  | Screenshot 2025-08-26 at 1 25 54 AM |

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Infrastructure change (CI configs, etc)
- [ ] Non-code change (docs, etc)
- [ ] None of the above: (explain here)

## Test Plan
The easiest way to test this is probably to clone the repo, install goreleaser, and build the binary locally. See "Building the CLI" in the [CONTRIBUTING.md file](https://github.com/pinecone-io/cli/blob/main/CONTRIBUTING.md#building-the-cli).

```
goreleaser build --single-target --snapshot --clean

# run via .dist using the proper path for your OS, you can remove the --single-target to build all available binaries

./dist/pc_darwin_arm64/pc auth configure --client-id <your-client-id> --client-secret <your-client-secret>

# OR use the browser login flow

./dist/pc_darwin_arm64/pc login

./dist/pc_darwin_arm64/pc organization list
./dist/pc_darwin_arm64/pc project list
./dist/pc_darwin_arm64/pc api-key list

# test the various command crud operations
# NOTE: some of the operations involving projects and api-keys will leverage the project that's currently targeted via pc auth target, and testing edge cases around this behavior would be helpful

```

### CODE REVIEW

File: internal/pkg/utils/configuration/state/context.go [hunk 1]
```diff
@@ -11,3 +13,19 @@ func GetTargetContext() *TargetContext {
 		Project: TargetProj.Get().Name,
 	}
 }
+
+func GetTargetOrgId() (string, error) {
```
Comment (austin-denoble) [inline internal/pkg/utils/configuration/state/context.go:17]: Moved these functions into this package from command specific packages.
